src/components/CarouselComponent/CarouselComponent.tsx

```diff
@@ -61,7 +63,31 @@ function CarouselComponent() {
     dataPack.length >= slidesNum()! && dataArr.push(dataPack)
   }
 
+  function handleCardClick(latitude: string, longitude: string) {
+    return function() {
+      // Here, perform the action you need with the latitude and longitude
+      console.log("Latitude:", latitude, "Longitude:", longitude);
+      
+      const lat = Number(latitude);
+      const lng = Number(longitude);
+      const zoomLevel = 12;
+      console.log("Latitude:", lat, "Longitude:", lng)
+      if (!isNaN(lat) && !isNaN(lng)) {
+        console.log(lat+","+lng)
+        centerCard(lat, lng, zoomLevel);
+        console.log(centerCard);
+        console.log("success");
+      } else {
+        console.error("Invalid latitude or longitude values");
+      
+      }
+    };
+  }
```

Comment:
This function can be implemented in a more readable and less complex way:

` const handleCardClick = (location: LatLngExpression) => {
    map?.flyTo(location)
  }`